x-pack/lib/config_management/elasticsearch_source.rb

```diff
@@ -50,6 +49,21 @@ def config_conflict?
         false
       end
 
+      # decide using system indices api (7.10+) or legacy api (< 7.10) base on elasticsearch server version
+      def pipeline_fetcher_factory
+        response = client.get("/")
+
+        if response["error"]
+          raise RemoteConfigError, "Cannot find elasticsearch version, server returned status: `#{response["status"]}`, message: `#{response["error"]}`"
+        end
+
+        logger.debug("Elasticsearch version ", response["version"]["number"])
```

Comment:
As it stands, this line won't output the value of `response["version"]["number"] - you can update this by either:
```logger.debug("Elasticsearch version {}", response["version"]["number"])```, 
```logger.debug("Elasticsearch version #{response['version']['number']}")``` or
```logger.debug("Elasticsearch version ", :version => response["version"]["number"])```

You might also want to add some context as to what is happening, such as "Reading configuration from Elasticsearch version..."